src/features/CartProductView/ui/CartProductView.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState } from "react";
+import {
+  Box,
+  Grid,
+  IconButton,
+  Paper,
+  type PaperProps,
+  Stack,
+  Typography,
+} from "@mui/material";
+import { PriceTag } from "../../../shared/ui/PriceTag";
+import AddCircleOutlineOutlinedIcon from "@mui/icons-material/AddCircleOutlineOutlined";
+import RemoveCircleOutlineOutlinedIcon from "@mui/icons-material/RemoveCircleOutlineOutlined";
+import DeleteOutlinedIcon from "@mui/icons-material/DeleteOutlined";
+import {
+  deleteIconButtonStyle,
+  attributeStyle,
+  counterStyle,
+  titleStyle,
+} from "./style";
+import { ThemeColors } from "../../../shared/constants/colors";
+import type { CartProductData } from "../model/types";
+import {
+  changeCartProductQuantity,
+  cartSlice,
+  getCurrentLineItem,
+} from "../../../entities/cart";
+import { getPriceValue } from "../../../shared/api/product";
+import { useAppDispatch } from "../../../shared/model/hooks";
+
+export type CartProductViewProps = PaperProps & {
+  data: CartProductData;
+};
+
+export function CartProductView({
+  data,
+  sx,
+  ...paperProps
+}: CartProductViewProps) {
+  const {
+    id,
+    title,
+    image,
+    price,
+    discountPrice,
+    color,
+    size,
+    quantity,
+    totalPrice,
+  } = data;
+
+  const [totalPriceValue, setTotalPriceValue] = useState(totalPrice);
+
+  const [counter, setCounter] = useState(quantity);
+
+  const [isButtonDisabled, setIsButtonDisabled] = useState(false);
+
+  const { updateTotalPrice } = cartSlice.actions;
+
+  const dispatch = useAppDispatch();
+
+  const updateProductQuantity = (action: "add" | "remove") => {
+    void (async () => {
+      setIsButtonDisabled(true);
+
+      const cart = await changeCartProductQuantity(id, action);
+
+      const currentLineItem = getCurrentLineItem(cart.lineItems, id);
+
+      setCounter(currentLineItem.quantity);
+
+      setTotalPriceValue(getPriceValue(currentLineItem.totalPrice));
+
+      const cartTotalPrice = getPriceValue(cart.totalPrice);
+
+      dispatch(updateTotalPrice(cartTotalPrice));
+
+      setIsButtonDisabled(false);
+    })();
```

Comment:
```suggestion
  const updateProductQuantity = (action: "add" | "remove") => {
    void (async () => {
      setIsButtonDisabled(true);

      const cart = await changeCartProductQuantity(id, action);
      const currentLineItem = getCurrentLineItem(cart.lineItems, id);

      setCounter(currentLineItem.quantity);
      setTotalPriceValue(getPriceValue(currentLineItem.totalPrice));

      const cartTotalPrice = getPriceValue(cart.totalPrice);

      dispatch(updateTotalPrice(cartTotalPrice));
      setIsButtonDisabled(false);
    })();
```

Что-то в духе. Если тут можно объединить все объявления переменной в одном блоке - сделай это.

Reply:
Исправил